### PR TITLE
Add github action to start closing stale issues

### PR DESCRIPTION
**Pull request by BenVillalobos**
Part 1/x to fix https://github.com/dotnet/msbuild/issues/6472

Let's reduce the energy required to keep up with some of the issues we get in this repo.

### Context
We should be strict when it comes to cutting out issues that don't get responses from customers. This change closes out issues after ~1.5 months of inactivity that require the author of an issue to respond.

### Changes Made
Created a github action to:
1. Mark issues stale after 30 days of inactivity
2. Close them two weeks after they've been marked stale
3. To start, **this will only affect issues marked as "Needs: Author Feedback"**

### For the dev team
When you respond to github issues and need authors to respond, please apply "Needs: Author Feedback". We can figure out how to scale this over time, but starting with a small subset of issues is the right move.

To avoid a situation where customers are forced to ["bump" their issues](https://github.com/actions/stale/issues/288), remove "Needs: Author Feedback" from the issue.

### CODE REVIEW

File: .github/workflows/stale.yml
Comment (rainersigwald) [inline .github/workflows/stale.yml:13]:
```suggestion
            This issue is marked as stale due because author feedback has been requested for 30 days with no response. Please respond within 5 days or this issue will be closed due to inactivity.
```

Comment (rainersigwald) [inline .github/workflows/stale.yml:13]: Is the 5/14 mismatch intentional?

File: .github/workflows/stale.yml
Comment (rainersigwald) [inline .github/workflows/stale.yml:16]: This seems unlikely to be generally helpful.

File: .github/workflows/stale.yml
Comment (rainersigwald) [inline .github/workflows/stale.yml:18]:
```suggestion
            This issue was closed due to inactivity. If you can still reproduce this bug, please comment with the requested information, detailed steps to reproduce the problem, or any other notes that might help in the investigation.
```

File: .github/workflows/stale.yml
Comment (rainersigwald) [inline .github/workflows/stale.yml:21]: Also exempt issues in the `backlog` milestone?

Comment (rainersigwald) [inline .github/workflows/stale.yml:21]: Should we turn on `remove-stale-when-updated`? I tend to think so: even if the OP has gone away someone else could help us chase an issue.

File: .github/workflows/stale.yml
Comment (rainersigwald) [inline .github/workflows/stale.yml:19]: Are we sure we want to do this for PRs too? That's an uncomfortable conversation in our current backlogged state :(

File: .github/workflows/stale.yml
Comment (rainersigwald) [inline .github/workflows/stale.yml:21]: Let's add a start date of now (or one month ago). We don't have to keep it forever but I'd like to not accidentally spam ourselves with a billion messages for very stale things before we know if we like the process.

File: .github/workflows/stale.yml
Comment (rainersigwald) [inline .github/workflows/stale.yml:10]: Is there a way to configure this to _add_ a label when it unstales an issue? What I really want to avoid is

1. User files
2. We ask for info
3. They respond
4. we miss it
5. Bot stales
6. Bot closes

Because that's super annoying.

**Review by rainersigwald — COMMENTED** [review]
Many comments, suggestions. Not 100% sure I want to go down this route but I can probably be convinced.

File: .github/workflows/stale.yml
Comment (japj) [reply .github/workflows/stale.yml:21]: Sometimes the original author has moved on to different things (e.g. switched projects or company) so tying updates to the original submitter might be tricky

File: .github/workflows/stale.yml
Comment (BenVillalobos) [reply .github/workflows/stale.yml:19]: I intentionally didn't set any label information for PR's, and by restricting to "Needs Author Feedback" (which should only be applied to issues?) it shouldn't consider PR's.

File: .github/workflows/stale.yml
Comment (BenVillalobos) [reply .github/workflows/stale.yml:16]: Saw something similar to this in another repo's stale yml file. They had this awesome section that said "if this still fails on latest/master..." which we can't tell people, so I can see why this isn't super helpful.

File: .github/workflows/stale.yml
Comment (BenVillalobos) [reply .github/workflows/stale.yml:10]: This is something I looked for but couldn't find support for. Created https://github.com/actions/stale/issues/460 that this PR should potentially be blocked on. I have no issue removing "Needs: Author Feedback" as issues are marked un-stale though.

File: .github/workflows/stale.yml
Comment (BenVillalobos) [reply .github/workflows/stale.yml:18]: Remove-stale-when-updated is defaulted to true, I don't think there's a way to specify "remove-stale-when-author-updated". I think it's fine to un-stale when anyone responds.

File: .github/workflows/stale.yml
Comment (BenVillalobos) [reply .github/workflows/stale.yml]: Was a mistake on my end. I modified your suggestion to fix that before committing.

File: .github/workflows/stale.yml
Comment (Forgind) [inline .github/workflows/stale.yml:21]: We might eventually want to make backlog fair game. That would be the semantic equivalent of "closed: low priority" from AzDO.

Also, it would be nice to add other milestones. I'm wondering if, rather than having a 16.11 milestone, a 17.0 milestone, etc., if we should have "current release," "last minor release," "penultimate minor release," and "last major release." Advantages: we could set it here without needing to update this, and we wouldn't have to update milestones on issues we didn't get to but are important. Downside: some issues should be targeted for a specific release, and they would silently move forward instead of staying where they should be.

File: .github/workflows/stale.yml
Comment (Forgind) [reply .github/workflows/stale.yml:17]: I think we should explicitly have a not-PRs part. There's no reason we _couldn't_ put Needs: Author Feedback on a PR, and if we want to make it more general later, we would want that eventually.

File: .github/workflows/stale.yml
Comment (Forgind) [inline .github/workflows/stale.yml:18]: Is this 14 days between "stale" and "closed"? So 44 days total?

Comment (BenVillalobos) [reply .github/workflows/stale.yml:18]: Yeah, wasn't thinking about a nice number. Rather, "one month until marked as stale", and "two weeks from then to close"

File: .github/workflows/stale.yml
Comment (rainersigwald) [reply .github/workflows/stale.yml:19]: I think the start date that got added is January 6 rather than June 1, right?

**Review by rainersigwald — APPROVED** [review]
LGTM after date tweak

File: .github/workflows/stale.yml
Comment (BenVillalobos) [reply .github/workflows/stale.yml:19]: I think so. Swapped over the day/month and removed the suffix.

File: .github/workflows/stale.yml
Comment (Forgind) [inline .github/workflows/stale.yml:4]: What does this mean?

File: .github/workflows/stale.yml
Comment (Forgind) [inline .github/workflows/stale.yml:17]: Might want to update these numbers again 😉

File: .github/workflows/stale.yml
Comment (Forgind) [inline .github/workflows/stale.yml:21]: Also exempt our other milestones? Just thinking a blanket no-milestones requirement makes sense.

File: .github/workflows/stale.yml
Comment (Forgind) [inline .github/workflows/stale.yml:23]: And then revert the label to check for.

**Review by Forgind — APPROVED** [review]
Changes look good! Though please revert the changes you used to test before merging. 🙂

**Comment by Forgind** [comment]
I heard you were thinking of doing this a different way—should we close this, or are you planning to modify it when finished?

**Comment by BenVillalobos** [comment]
@Forgind The current plan is to use the fabric bot for rapid testing. We'll likely want to modify the numbers, messages, etc. in the short term and I'd prefer not to have a PR out for every little change. Once we're happy with it, we can update this PR to reflect the bot and get this merged in.

Since you prefer to have minimal PR's I'll close this until either:

1. We get fabric bot enabled, test it, and are happy with the setup
2. We can't get it enabled after a week or two and we have no choice but to test through the GH action.

File: .github/workflows/stale.yml
Comment (BenVillalobos) [reply .github/workflows/stale.yml:21]: Good idea, commenting to remind myself when this opens back up.

File: .github/workflows/stale.yml
Comment (BenVillalobos) [reply .github/workflows/stale.yml:17]: Yeah reverting the `debug mode` commit will come after merging and testing it out.

File: .github/workflows/stale.yml
Comment (BenVillalobos) [reply .github/workflows/stale.yml:4]: It's arbitrarily set to 7PM, but should probably be `00 12 * * *` (noon). (NTS when this is picked back up)

**Comment by Forgind** [comment]
Sounds good to me! Thanks. And nice label.

**Pull request closed without merging**
